### PR TITLE
Refine the subprocess management

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Maybe will solve #834 . And it's a preparation for new iochaos (which also needs subprocess management)

### What is changed and how does it work?

By setting `prctl(PR_SET_PDEATHSIG, SIGTERM)` for `nsenter` and the subprocess of `nsenter`. As the `nsenter` will fork when we enter a new pid namespace, and nothing is done for killing subprocess or suiciding when the parent dies.

For example: 

If you call `nsenter -p/proc/.../ns/pid bash -c "while true; do sleep 1; date; done"` then even you kill the nsenter process, the subprocess of it will continue running until it gets killed. The suicide program is used to make sure that the subprocess will be terminated when its parent died.

I'm not sure this method is 100% reliable, but half a loaf is better than none.

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

1. Stop the chaos-daemon will stop all stress-ng process.

2. Apply and recover the StressChaos works normally.

### CODE REVIEW

File: api/v1alpha1/stresschaos_types.go [hunk 1]
```diff
@@ -119,9 +119,6 @@ type StressInstance struct {
 	// UID is the instance identifier
 	// +optional
 	UID string `json:"uid"`
-	// StartTime specifies when the instance starts
-	// +optional
-	StartTime *metav1.Time `json:"startTime"`
```
Comment (cwen0) [inline api/v1alpha1/stresschaos_types.go:124]: Why delete `StartTime`? This `StartTime` was used to avoid the PID was reused.

Comment (YangKeao) [reply api/v1alpha1/stresschaos_types.go:124]: Oops. It's really a mistake. Checking whether the target is the child process of chaos-daemon cannot replace this start-time check.

I will fix it right now.

File: pkg/chaosdaemon/util.go [hunk 2]
```diff
@@ -212,59 +213,111 @@ func GetNsPath(pid uint32, typ nsType) string {
...
-		f := c.(func(context.Context, []nsOption, string, ...string) *exec.Cmd)
-		return f(ctx, options, cmd, args...)
+func (b *processBuilder) EnableSuicide() *processBuilder {
+	b.suicide = true
+
+	return b
+}
+
+func (b *processBuilder) Build(ctx context.Context) *exec.Cmd {
+	// The call routine is suicide -> pause -> nsenter --(fork)-> suicide -> process
```
Comment (cwen0) [inline pkg/chaosdaemon/util.go:270]: s/suicide -> pause /pause->suicide/g

Comment (YangKeao) [reply pkg/chaosdaemon/util.go]: Fixed

**Review by cwen0 — APPROVED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

**Pull request merged**